packages/k8s/src/k8s/index.ts

```diff
@@ -516,27 +516,34 @@ class BackOffManager {
 export function containerPorts(
   container: ContainerInfo
 ): k8s.V1ContainerPort[] {
-  // 8080:8080/tcp
-  const portFormat = /(\d{1,5})(:(\d{1,5}))?(\/(tcp|udp))?/
-
   const ports: k8s.V1ContainerPort[] = []
   for (const portDefinition of container.portMappings) {
-    const submatches = portFormat.exec(portDefinition)
-    if (!submatches) {
+    const portProtoSplit = portDefinition.split('/')
+    if (portProtoSplit.length > 2) {
       throw new Error(
-        `Port definition "${portDefinition}" is in incorrect format`
+        `port definition should be in format [port(s)]/[proto], got ${portDefinition}`
       )
     }
+
     const port = new k8s.V1ContainerPort()
-    port.hostPort = Number(submatches[1])
-    if (submatches[3]) {
-      port.containerPort = Number(submatches[3])
+    port.protocol =
+      portProtoSplit.length === 2 ? portProtoSplit[1].toUpperCase() : 'TCP'
+
+    const portSplit = portProtoSplit[0].split(':')
+    if (portSplit.length > 2) {
+      throw new Error('ports should have at most one ":" separator')
     }
-    if (submatches[5]) {
-      port.protocol = submatches[5].toUpperCase()
+    if (!Number(portSplit[0])) {
+      throw new Error('port specification is invalid')
+    }
+
+    if (portSplit.length === 1) {
+      port.containerPort = Number(portSplit[0])
```

Comment:
can we validate these are in the correct port range and otherwise throw an error

0 < x < 65536

Reply:
Would also be good to add unit tests for those scenarios!